advanced_source/static_quantization_tutorial.rst

```diff
@@ -0,0 +1,665 @@
+
+
+    def accuracy(output, target, topk=(1,)):  
+        """Computes the accuracy over the k top predictions for the specified values of k"""  
+        with torch.no_grad(): 
+            maxk = max(topk)  
+            batch_size = target.size(0) 
+
+            _, pred = output.topk(maxk, 1, True, True)  
+            pred = pred.t() 
+            correct = pred.eq(target.view(1, -1).expand_as(pred)) 
+
+            res = []  
+            for k in topk:  
+                correct_k = correct[:k].reshape(-1).float().sum(0, keepdim=True)  
+                res.append(correct_k.mul_(100.0 / batch_size))  
+            return res  
+
+
+    def evaluate(model, criterion, data_loader, neval_batches): 
+        model.eval()  
+        top1 = AverageMeter('Acc@1', ':6.2f') 
+        top5 = AverageMeter('Acc@5', ':6.2f') 
+        cnt = 0 
+        with torch.no_grad(): 
+            for image, target in data_loader: 
+                output = model(image) 
+                loss = criterion(output, target)  
+                cnt += 1  
+                acc1, acc5 = accuracy(output, target, topk=(1, 5))  
+                print('.', end = '')  
+                top1.update(acc1[0], image.size(0)) 
+                top5.update(acc5[0], image.size(0)) 
+                if cnt >= neval_batches:  
+                     return top1, top5  
+
+        return top1, top5 
+
+    def load_model(model_file): 
+        model = MobileNetV2() 
+        state_dict = torch.load(model_file) 
+        model.load_state_dict(state_dict) 
+        model.to('cpu') 
+        return model  
+
+    def print_size_of_model(model): 
+        torch.save(model.state_dict(), "temp.p")  
+        print('Size (MB):', os.path.getsize("temp.p")/1e6)  
+        os.remove('temp.p') 
+
+3. Define dataset and data loaders  
+----------------------------------  
+ 
+As our last major setup step, we define our dataloaders for our training and testing set. 
+ 
+ImageNet Data 
+^^^^^^^^^^^^^ 
+
+Please download imagenet dataset by following the instructions at: TBD. The rest of the tutorial assumes that you have imagenet_1k.zip which contains the sample imagenet dataset.
+
+Unzip imagenet_1k.zip to `data` folder.
+
+For this tutorial to run, we download this data and move it to the right place using  
+`these lines <https://github.com/pytorch/tutorials/blob/master/Makefile#L97-L98>`_  
+from the `Makefile <https://github.com/pytorch/tutorials/blob/master/Makefile>`_. 
+
+To run the code in this tutorial using the entire ImageNet dataset, on the other hand, you could download 
+the data using ``torchvision`` following  
+`here <https://pytorch.org/docs/stable/torchvision/datasets.html#imagenet>`_. For example,  
+to download the training set and apply some standard transformations to it, you could use:  
+
+.. code:: python  
+
+    import torchvision  
+    import torchvision.transforms as transforms 
+
+    imagenet_dataset = torchvision.datasets.ImageNet( 
```

Comment:
This part looks incomplete. We are defining an imagenet_dataset first, but not using it later. Also, the transforms here do not match the transforms defined below.

Reply:
this is just used to download training data I think?

Also I didn't see how the transformation is different? can you elaborate?

Reply:
imagenet_dataset is not being used later, is that the intent?.

Reply:
I think this is used to download data